tecken/ext/gcs/storage.py

```diff
@@ -0,0 +1,146 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from io import BufferedReader
+import threading
+from typing import Optional
+from urllib.parse import quote
+
+from django.conf import settings
+
+from google.api_core.client_options import ClientOptions
+from google.api_core.exceptions import ClientError, NotFound
+from google.cloud import storage
+
+from tecken.libstorage import ObjectMetadata, StorageBackend, StorageError
+
+
+class GCSStorage(StorageBackend):
+    """
+    An implementation of the StorageBackend interface for Google Cloud Storage.
+    """
+
+    def __init__(
+        self,
+        bucket: str,
+        prefix: str,
+        try_symbols: bool = False,
+        endpoint_url: Optional[str] = None,
+    ):
+        self.bucket = bucket
+        self.prefix = prefix
+        self.try_symbols = try_symbols
+        self.endpoint_url = endpoint_url
+        self.clients = threading.local()
+        # The Cloud Storage client doesn't support setting global timeouts for all requests, so we
+        # need to pass the timeout for every single request. the default timeout is 60 seconds for
+        # both connecting and reading from the socket.
+        self.timeout = (settings.S3_CONNECT_TIMEOUT, settings.S3_READ_TIMEOUT)
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} gs://{self.bucket}/{self.prefix}>"
```

Comment:
This is what this currently looks like in the GCP stage env:
```
[<GCSStorage gs://moz-fx-symbols-nonprod-stage-tecken/v1>,
 <GCSStorage gs://moz-fx-symbols-nonprod-stage-tecken/try/v1>,
 <S3Storage s3://org.mozilla.crash-stats.symbols-public/v1>,
 <S3Storage s3://org.mozilla.crash-stats.symbols-public/try/v1>]
```
Since there is `try/` in the URL, I thought that would be clear enough in practice, but of course that doesn't stricly imply that we actually set the `try_symbols` flag correctly.

Do you think the `try/` is good enough, or should we include the flag as well?

Reply:
We talked about this in a Zoom meeting, and we decided to include the try flag explicitly as well.